### PR TITLE
harmonize convergence checks

### PR DESCRIPTION
**Pull request by FFroehlich**
use the same routines for newton and steadystate, newton still doesn't check sensitivity convergence

### CODE REVIEW

File: src/steadystateproblem.cpp [hunk 1]
```diff
@@ -16,6 +16,8 @@
 #include <memory>
 #include <cvodes/cvodes.h>
 
+#define CONV_T = RCONST(1.0)
```
Comment (dweindl) [inline src/steadystateproblem.cpp]:
```suggestion
constexpr realtype convergence_threshold = 1.0;
```
?

**Review by dweindl — APPROVED** [review]
:+1:

File: src/steadystateproblem.cpp [hunk 2]
```diff
@@ -433,41 +441,36 @@ realtype SteadystateProblem::getWrmsNorm(const AmiVector &x,
...
-                solver->getRelativeToleranceSteadyStateSensi(), ewt_);
-            converged = wrms_ < RCONST(1.0);
-        }
+        if (wrms >= conv_thresh)
+            break;
+        model->fsxdot(t_, x_, dx_, ip, sx_[ip], dx_, xdot_);
+        wrms = getWrmsNorm(sx_[ip], xdot_, atol_sensi_, rtol_sensi_, ewt_);
     }
-    return converged;
+    return wrms;
```
Comment (plakrisenko) [inline src/steadystateproblem.cpp:473]: wrms for the last parameter is returned?

Comment (FFroehlich) [reply src/steadystateproblem.cpp:473]: > wrms for the last parameter is returned?

for the current `x_`, yes.

File: src/steadystateproblem.cpp [hunk 3]
```diff
@@ -433,41 +441,36 @@ realtype SteadystateProblem::getWrmsNorm(const AmiVector &x,
...
     for (int ip = 0; ip < model->nplist(); ++ip) {
-        if (converged) {
-            sx_ = solver->getStateSensitivity(t_);
-            model->fsxdot(t_, x_, dx_, ip, sx_[ip], dx_, xdot_);
-            wrms_ = getWrmsNorm(
-                sx_[ip], xdot_, solver->getAbsoluteToleranceSteadyStateSensi(),
-                solver->getRelativeToleranceSteadyStateSensi(), ewt_);
-            converged = wrms_ < RCONST(1.0);
-        }
+        if (wrms >= conv_thresh)
```
Comment (plakrisenko) [inline src/steadystateproblem.cpp:468]: wouldn't it break at the first step because INFINITY >= 1?

Comment (FFroehlich) [reply src/steadystateproblem.cpp:468]: yes, good point!

File: src/steadystateproblem.cpp (hunk 2 above)
Comment (plakrisenko) [reply src/steadystateproblem.cpp:473]: shouldn't it be an aggregation of all nplist `wrms`es?

Comment (FFroehlich) [reply src/steadystateproblem.cpp:473]: ah, now I understand the question. We want the maximum of the `wrms` over all `ip`, but if any of them is larger than the convergence threshold, we can stop as we know we aren't converged yet. Will fix and document.

**Review by plakrisenko — APPROVED** [review]

**Pull request merged**